src/ExchangeSharp/API/Exchanges/Coinmate/ExchangeCoinmateAPI.cs

```diff
@@ -0,0 +1,301 @@
+
+			return orders.Select(x => new ExchangeOrderResult
+			{
+				Amount = x.Amount,
+				ClientOrderId = x.ClientOrderId?.ToString(),
+				IsBuy = x.Type == "BUY",
+				MarketSymbol = x.CurrencyPair,
+				OrderDate = CryptoUtility.ParseTimestamp(x.Timestamp, TimestampType.UnixMilliseconds),
+				OrderId = x.Id.ToString(),
+				Price = x.Price,
+
+			}).ToArray();
+		}
+
+		protected override async Task<ExchangeDepositDetails> OnGetDepositAddressAsync(string currency, bool forceRegenerate = false)
+		{
+			var payload = await GetNoncePayloadAsync();
+			var currencyName = GetCurrencyName(currency);
+			var addresses = await MakeCoinmateRequest<string[]>($"/{currencyName}DepositAddresses", payload, "POST");
+
+			return new ExchangeDepositDetails
+			{
+				Address = addresses.FirstOrDefault(),
+				Currency = currency,
+			};
+		}
+
+		protected override async Task<ExchangeWithdrawalResponse> OnWithdrawAsync(ExchangeWithdrawalRequest withdrawalRequest)
+		{
+			var payload = await GetNoncePayloadAsync();
+			var currencyName = GetCurrencyName(withdrawalRequest.Currency);
+
+			payload["amount"] = withdrawalRequest.Amount;
+			payload["address"] = withdrawalRequest.Address;
+			payload["amountType"] = withdrawalRequest.TakeFeeFromAmount ? "NET" : "GROSS";
+
+			var id = await MakeCoinmateRequest<long?>($"/{currencyName}Withdrawal", payload, "POST");
+
+			return new ExchangeWithdrawalResponse
+			{
+				Id = id?.ToString(),
+				Success = id != null
+			};
+		}
+
+		protected override async Task ProcessRequestAsync(IHttpWebRequest request, Dictionary<string, object> payload)
+		{
+			if (CanMakeAuthenticatedRequest(payload))
+			{
+				if (string.IsNullOrWhiteSpace(ClientId))
+				{
+					throw new APIException("Client ID is not set for Coinmate");
+				}
+
+				var apiKey = PublicApiKey.ToUnsecureString();
+				var messageToSign = payload["nonce"].ToStringInvariant() + ClientId + apiKey;
+				var signature = CryptoUtility.SHA256Sign(messageToSign, PrivateApiKey.ToUnsecureString()).ToUpperInvariant();
+				payload["signature"] = signature;
+				payload["clientId"] = ClientId;
+				payload["publicKey"] = apiKey;
+				await CryptoUtility.WritePayloadFormToRequestAsync(request, payload);
+			}
+		}
+
+		private async Task<T> MakeCoinmateRequest<T>(string url, Dictionary<string, object> payload = null, string method = null)
+		{
+			var response = await MakeJsonRequestAsync<CoinmateResponse<T>>(url, null, payload, method);
+
+			if (response.Error)
+			{
+				throw new APIException(response.ErrorMessage);
+			}
+
+			return response.Data;
+		}
+
+		private string GetCurrencyName(string currency)
+		{
+			return currency.ToUpper() switch
+			{
+				"BTC" => "bitcoin",
+				"LTC" => "litecoin",
+				"BCH" => "bitcoinCash",
+				"ETH" => "ethereum",
+				"XRP" => "ripple",
+				"DASH" => "dash",
+				"DAI" => "dai",
+				_ => throw new NotImplementedException("Unsupported currency")
+			};
+		}
+	}
```

Comment:
Could you add a partial class to the end like the other exchanges, such as:
https://github.com/jjxtra/ExchangeSharp/blob/6e327ab82aff0db6cdc0846bbdc651324e525e8b/src/ExchangeSharp/API/Exchanges/Coinbase/ExchangeCoinbaseAPI.cs#L818